src/test/scala/org/neo4j/spark/DataSourceReaderTSE.scala

```diff
@@ -1096,6 +1097,25 @@ class DataSourceReaderTSE extends SparkConnectorScalaBaseTSE {
     assertEquals(2, df.count())
   }
 
+  @Test
+  def testShouldPassTheScriptResult(): Unit = {
+    val df = ss.read
+      .format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("script", "RETURN 'foo' AS val")
```

Comment:
Nice.  Just a question on this point:  While partitions=2 here, does script get executed twice, once in each partition, or once upfront and then passed to each partition?  I think it gets executed twice, right?

Reply:
No, the query gets executed just once on the Spark Driver program an then the result is shared across partitions